z390test/src/test/groovy/org/z390/test/RunTestOption.groovy

```diff
@@ -0,0 +1,109 @@
+package org.z390.test
+
+import org.junit.jupiter.api.Test
+
+class RunTestOption extends z390Test {
+
+    @Test
+    void test_TESTOPT_OK() {
+        /**
+         * test 1 - should assemble okay
+         */
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${basePath("rt", "opt", "testopt0.OPT")}")
+        this.printOutput()
+        assert rc == 0
+    }
+
+    @Test
+    void test_TESTOPT1_invalidOption() {
+        /**
+         * test 2 - should assemble with error for invalid option
+         */
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${basePath("rt", "opt", "testopt1.OPT")}")
+        this.printOutput()
+        assert rc == 16
+        assert this.fileData['ERR'] =~ /invalid options -\s{2}chksrc\(2X\)/
+        assert this.stdout =~ /invalid options -\s{2}chksrc\(2X\)/
+    }
+
+    @Test
+    void test_TESTOPT2L_validReference() {
+        /**
+         * test 3 - should assemble okay with valid reference
+         */
+        var testOpt0FileContents = """
+* Test with a valid option
+  chksrc(2)  * Check sources for illegal characters
+"""
+        var innerOptionFilename = createTempFile("testopt0.OPT", testOpt0FileContents)
+        var testOpt2LFileContents = """
+* Test with a valid subfile
+  chksrc(2)  * Check sources for illegal characters
+@${innerOptionFilename}
+"""
+        var OptionFilename = createTempFile("testopt2L.OPT", testOpt2LFileContents)
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${OptionFilename}")
+        this.printOutput()
+        assert rc == 0
+        assert fileData['PRN'] =~ /testopt2L.OPT=/
+        assert fileData['PRN'] =~ /testopt0.OPT=/
+    }
+
+    @Test
+    void test_TESTOPT3L_incorrectReference() {
+        /**
+         * test 4 - should assemble with error for incorrect reference
+         */
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${basePath("rt", "opt", "testopt3L.OPT")}")
+        this.printOutput()
+        assert rc == 16
+        assert this.stdout =~ /TZ390E abort error 21 - invalid options -\s{2}@rt\/opt\/testoptX/
+        assert this.fileData['ERR'] =~ /TZ390E abort error 21 - invalid options -\s{2}@rt\/opt\/testoptX/
+    }
+
+    @Test
+    void test_TESTOPT4L_selfReference() {
+        /**
+         * test 5 - should assemble with error for self-reference
+         */
+        var OptionFileContents = """
+* Test with a self-reference
+  chksrc(2)  * Check sources for illegal characters
+@{{fullFileName}}"""
```

Comment:
Don't understand this. Enclosing double braces? What is fullFileName? I'll have to trust you on this one.

Reply:
OK.. this was one of the challenges of this test.. I needed to self reference the temp filename.. Its a chicken and egg with the temp file function.. you don't know the file name until its created, but the file contents contain the generated filename.. The function provides the special {{fullFileName}} placeholder so that the function inserts the name of the temp file at the location of {{fullFileName}}.
I have added a description of this behaviour in the README.md file for the createTempFile function:

  If you need to include the name of the generated file in the file, use `{{fullFileName}}` placeholder in
  file contents and the function will replace this with the actual filename created.